source/agora/consensus/Ledger.d

```diff
@@ -91,21 +91,19 @@ public class NodeLedger : Ledger
 
         Params:
             params = the consensus-critical constants
-            engine = script execution engine
-            utxo_set = the set of unspent outputs
+            database = State database
             storage = the block storage
+            engine = script execution engine
             enroll_man = the enrollmentManager
             pool = the transaction pool
-            fee_man = the FeeManager
             onAcceptedBlock = optional delegate to call
                               when a block was added to the ledger
 
     ***************************************************************************/
 
     public this (immutable(ConsensusParams) params,
-        Engine engine, UTXOSet utxo_set, IBlockStorage storage,
-        EnrollmentManager enroll_man, TransactionPool pool,
-        FeeManager fee_man,
+        ManagedDatabase database, IBlockStorage storage,
+        Engine engine, EnrollmentManager enroll_man, TransactionPool pool,
```

Comment:
We could do the same for `engine` as well, it is not stateful and pretty lightweight too. There is no reason to share it.

Reply:
Engine could even be disposible, created only in the scope that it is needed.